init.zsh

```diff
@@ -7,8 +7,12 @@ if [[ -z "$ANTIGEN_HS_MY" ]]; then
     ANTIGEN_HS_MY="$ANTIGEN_HS_HOME/../MyAntigen.hs"
 fi
 
+antigen-hs-init () {
+  cd "$ANTIGEN_HS_HOME" && stack build
+}
+
 antigen-hs-compile () {
-  runghc -i"$ANTIGEN_HS_HOME/" -- "$ANTIGEN_HS_MY"
+  stack exec -- runghc -i"$ANTIGEN_HS_HOME/" -- "$ANTIGEN_HS_MY"
```

Comment:
I think `stack exec` approach is great. Just either make something like `antigen-hs-compile-using-stack` OR make `antigen-hs-compile` read envvar `ANTIGENHS_USE_STACK`. Both of those ways retain an easy path for most users but makes clean possible too.